pkg/k8s/third_party.go

```diff
@@ -66,7 +68,14 @@ func (r *CiliumNetworkPolicy) Parse() (api.Rules, error) {
 		if retRule.EndpointSelector.LabelSelector.MatchLabels == nil {
 			retRule.EndpointSelector.LabelSelector.MatchLabels = map[string]string{}
 		}
-		retRule.EndpointSelector.LabelSelector.MatchLabels[labels.LabelSourceK8sKeyPrefix+PodNamespaceLabel] = namespace
+
+		userNamespace, ok := retRule.EndpointSelector.LabelSelector.MatchLabels[labels.LabelSourceK8sKeyPrefix+PodNamespaceLabel]
+		if ok && userNamespace != namespace {
+			log.Warningf("k8s: CiliumNetworkPolicy %s/%s contains illegal namespace match '%s' in EndpointSelector."+
+				" EndpointSelector always applies in namespace of the policy resource, removing namespace match '%s'.",
+				r.Metadata.Namespace, r.Metadata.Name, userNamespace, userNamespace)
+			retRule.EndpointSelector.LabelSelector.MatchLabels[labels.LabelSourceK8sKeyPrefix+PodNamespaceLabel] = namespace
```

Comment:
This looks wrong. You always want to write the label but omit a warning if there was a mismatch. The assignment should be outside of the conditional.